repair/repair_task.hh

```diff
@@ -47,6 +47,20 @@ protected:
     future<> run() override;
 };
 
+class data_sync_repair_task_impl : public root_repair_task_impl {
+private:
+    dht::token_range_vector _ranges;
+    std::unordered_map<dht::token_range, repair_neighbors> _neighbors;
+    streaming::stream_reason _reason;
+    std::optional<tasks::task_id> _ops_uuid;
+public:
+    // TODO Deexie null task id instead of optional
```

Comment:
1. when is it set to null?
2. should it be a member of `root_repair_task_impl` so to be provided as a parent task_id?

Reply:
1. It is set to null for each data sync repair except for `removenode`. The only reason I can see behind that optional is a condition:
https://github.com/scylladb/scylladb/blob/6b180bf2c7830a5930ffcc87a9e4cf3a25db9092/repair/repair.cc#L450-L453
Where it is clearer that we cast it to bool on lhs and then compare the value on rhs than if we had: 
```